lib/3scale/backend/worker_async.rb

```diff
@@ -68,6 +68,10 @@ def schedule_jobs
           # unblocks when there are new jobs or when .close() is called
           job = @jobs.pop
 
+          # The queue is closed, no more jobs are going to be pushed, so
+          # shutdown.
+          shutdown if @jobs.closed?
```

Comment:
Can `job` be a valid job and in this point have `@jobs.closed?` be truthy? My suspicion is that this case is not possible, but I wanted to check with you.

Reply:
Hmm actually I think it's possible. I think `job` could be a valid one, and the other thread could call `@jobs.close` before this one calls `@jobs.closed?`

Reply:
I think we need this instead:
```

Reply:
Then the alternative question is: can `job` be `nil` and `@jobs.closed` be falsey? :D

Reply:
`Queue#pop` blocks until there's a new element in the queue or the queue is closed. So, if `job` is nil, the queue is closed.

Reply:
Ok, push the fix then and r=me.

Reply:
Fixed :rocket: